### PR TITLE
Update ConfigApplyRequest and ConfigUploadRequest protos

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Update ConfigApplyRequest and ConfigUploadRequest protos to remove duplicate fields.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: api/grpc/mpi/v1/files.proto [hunk 1]
```diff
@@ -49,6 +49,8 @@ message UpdateOverviewRequest {
     mpi.v1.MessageMeta message_meta = 1;
     // The file overview of an instance
     FileOverview overview = 2;
+    // The config version of the overview you are updating
+    ConfigVersion config_version = 3;
```
Comment (cohix) [inline api/grpc/mpi/v1/files.proto:53]: This part is fine

File: api/grpc/mpi/v1/files.proto [hunk 2]
```diff
@@ -66,8 +68,6 @@ message ConfigVersion {
 message FileOverview {
     // A list of files
     repeated File files = 1;
-    // The configuration version of the current set of files
-    ConfigVersion config_version = 2;
```
[the remark this answers is not in the document]
Comment (oliveromahony) [reply api/grpc/mpi/v1/files.proto:70]: We discussed not removing ConfigVersion here. I think originally, we thought that the FileOverview would represent other payloads that weren't specific to configuration. So we wanted to remove the Config specific fields. However, on reflection this message is very focused on Configuration. Therefore, keeping the ConfigVersion makes sense. I think the FileOverview name isn't specific enough though and probably lead to the above thought process. We should at least make a detailed comment about what this message is, if not rename it. What do you think?

**Comment by dhurley** [comment]
@cohix @oliveromahony Based on the feedback I reverted the changes to the `FileOverview` message and instead updated the `ConfigApplyRequest` & `ConfigUploadRequest` message to remove duplicate fields and make the `FileOverview` a required field.

**Comment by oliveromahony** [comment]
> @cohix @oliveromahony Based on the feedback I reverted the changes to the `FileOverview` message and instead updated the `ConfigApplyRequest` & `ConfigUploadRequest` message to remove duplicate fields and make the `FileOverview` a required field.

makes sense

**Review by cohix — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Review by aphralG — APPROVED** [review]

**Pull request merged**